gateway/ln-gateway/src/rpc/lnrpc_client.rs

```diff
@@ -101,3 +101,26 @@ impl ILnRpcClient for NetworkLnRpcClient {
         unimplemented!()
     }
 }
+
+/// A generic factory trait for creating `DynLnRpcClient` instances.
+#[async_trait]
+pub trait ILnRpcClientFactory: Debug {
```

Comment:
For the `NetworkLnRpcClient` which sends rpc requests over the wire, we may need a way of spawning a new client for every actor / actor rpc scenario because each rpc call is actually blocking on the client [ _read somewhere, looking for the right docs for reference_ ]

- If this is the case, we'd end up with a perf bottleneck, situation similar to #1577 
- It's okay to start off simple, with a single `DynLnRpcClient` instantiated and provided to gatewayd at boot. The factory pattern might be a solution down the road if we observe issues with the rpc client perf.